src/RSAccessor/RSPortalAccessorV2.cs

```diff
@@ -101,6 +101,41 @@ public void AddToCatalogItems(CatalogItem item)
             return item.Path;
         }
 
+        public void UpdateDataSourceCredentials(
+            string path,
+            string username,
+            string password,
+            bool isWindowsCredentials)
+        {
+            var ctx = CreateContext();
+            var report = ctx.CatalogItemByPath(path).Expand("DataSources").GetValue();
+
+            var powerBiReport = report as PowerBIReport;
+            if (powerBiReport != null)
+            {
+                foreach (var dataSource in powerBiReport.DataSources)
```

Comment:
It didn't throw any exception when I was testing it but I will still add some checks for this. Thanks!